### PR TITLE
bpo-40769: allow extra surrounding parentheses for invalid annotated assignment rule

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue40769

### CODE REVIEW

**Review by lysnikolaou — APPROVED** [review]
LGTM! Thanks. (You will probably need to regenerate parse.c)

File: Lib/test/test_peg_parser.py [hunk 1]
```diff
@@ -624,7 +624,11 @@ def f():
     ("True = 1", "cannot assign to True"),
     ("a() = 1", "cannot assign to function call"),
     ("(a, b): int", "only single target (not tuple) can be annotated"),
+    ("((a, b)): int", "only single target (not tuple) can be annotated"),
+    ("(()): int", "only single target (not tuple) can be annotated"),
     ("[a, b]: int", "only single target (not list) can be annotated"),
+    ("([a, b]): int", "only single target (not list) can be annotated"),
+    ("([]): int", "only single target (not list) can be annotated"),
```
Comment (lysnikolaou) [inline Lib/test/test_peg_parser.py]: On a second thought, could you add those to `test_syntax.py` as well?

**Comment by lysnikolaou** [comment]
@isidentical @pablogsal Friendly ping.

**Comment by isidentical** [comment]
rebased to upstream

**Comment by isidentical** [comment]
I've rebased again, CC: @pablogsal

**Review by pablogsal — APPROVED** [review]

**Pull request merged**